src/Infrastructure/MySQL/UserRepository.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace App\Infrastructure\MySQL;
+
+use App\Domain\Doctrine\DynamicConnection as Connection;
+use App\Domain\User;
+use App\Domain\UserRole;
+
+class UserRepository
+{
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    private const PROPERTY_NAME = 'user';
+
+    public function __construct(Connection $connection)
+    {
+        $this->connection = $connection;
+    }
+
+    public function inject(User $user, array $properties, array $injectedIds): array
+    {
+        $ids = [];
+
+        $yamlSubKey = match ($user->getRole()) {
+            UserRole::Administrator => 'administrators',
+            UserRole::Editor => 'editors',
+            UserRole::User => 'users',
+        };
+
+        $count = $properties[self::PROPERTY_NAME][$yamlSubKey];
+
+        $result = $this->connection->executeQuery('SELECT MAX(contact_id) AS max FROM contact');
+        $firstId = ((int) $result->fetchAssociative()['max']) + 1;
+        $maxId = $firstId + $count;
+
+        $baseQuery = 'INSERT INTO contact ' .
+            '(contact_id, contact_name, contact_alias, contact_email, contact_lang, contact_activate, contact_admin, contact_oreon) ' .
+            'VALUES ';
+        $valuesQuery = '';
+
+        $name = $user->getName() . '_';
+        $alias = $name;
+        $isAdmin = $user->getRole() === UserRole::Administrator ? '1' : '0';
+        $password = password_hash('centreon', PASSWORD_BCRYPT);
+
+        $insertCount = 0;
+        $userIndex = 0;
+        for ($i = $firstId; $i < $maxId; $i++) {
+            $ids[] = $i;
+            $insertCount++;
+            $userIndex++;
+            $valuesQuery .= '(' .
+                $i . ',' .
+                '"' . $name . $userIndex . '",' .
+                '"' . $alias . $userIndex . '",' .
+                '"' . $name . $userIndex . '@localhost",' .
+                '"en_US.UTF-8",' .
+                '"1",' .
+                '"' . $isAdmin . '",' .
+                '"1"' .
+                '),';
+            if ($insertCount === 50000) {
```

Comment:
It is safer to base this on the length of the request
`SHOW VARIABLES LIKE 'max_allowed_packet';`

Reply:
I have created ticket MON-21454 to handle this in all repositories